### ISSUE TITLE
Notify Webhook when ETL is finished to also index RPB

### ISSUE THREAD

**Issue by dr0i** [hbz/lobid-resources#1469]
After a newly created index the [RPB data](https://github.com/hbz/rpb) must be reindexed. This will be implemented in a first step by "guessing" a timeframe with enough safety distance for cron scheduling. The better approach is to call a RPB  Webhook listener to start indexing when ETL is actually finished.

If https://github.com/hbz/lobid-resources/issues/1468 is implemented we can reuse that for calling the RPB Webhook.

**Comment by dr0i** [comment]
Hm, after finishing the ETL of lobid-resources atm the alias is switched automatically. I.e., even after triggering the RPB indexing *directly* after the creation of the new lobid-resources index the RPB data will be missing in the index for at least the time the RPB indexing needs (atm ~10m).
Two solutions come into mind:

a) calling a webhook listener shortly after *starting* ETL (not  ETL is *finished*), providing the index-name so indexing can happen in parallel
b) set `webhook.alma.basedump.switch.automaticall` to `false`, trigger ETL of RPB and _then_ switch index alias

This just for background, a new issue should be created if RPB becomes productive.
First things first - solving this issue here would shrink the time of unavailable RPB data to `10m`, while without that we can expect a time frame of up to `6h` where the data will be missing.

**Comment by blackwinter** [comment]
I don't really know anything about your situation, but have you maybe considered maintaining separate indexes and accessing them through a common alias? Then you can update them independently.

**Comment by dr0i** [comment]
Thx @blackwinter this sounds like a wonderful idea! I admit to have not done this before so ... behaves e.g. the ranking like it would be _one_ index?

**Comment by blackwinter** [comment]
> behaves e.g. the ranking like it would be _one_ index?

Not exactly, no. If that's important to you, you might want to look into the `dfs_query_then_fetch` [search type parameter](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-request-search-type.html).

In DigiBib, we're doing this on a (semi-)regular basis with multiple indexes (without `dfs_query_then_fetch`, though).

**Comment by acka47** [comment]
I think, we can also close this issue for now as we switched to a separate index for RPB, see https://github.com/hbz/rpb/issues/21#issuecomment-1774636736

**Comment by dr0i** [comment]
Yes - I would be happier to have something like an "on hold" because it seems that you will use _one_ index one day. We might stumble on the problem described here and find this old issue an reopen it. Curious if this will happen (and when).
Closing for now.
